.github/workflows/check.yaml

```diff
@@ -52,6 +52,23 @@ jobs:
     - name: Run unit tests
       run: tox -e unit
 
+    - name: Check for unit test coverage file
+      id: check_unit_coverage
+      run: |
+        if [ -f tests/report/coverage-unit.xml ]; then
+          echo "exists=true" >> $GITHUB_OUTPUT
+        else
+          echo "exists=false" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Upload Unit Test Coverage Artifact
+      if: ${{ steps.check_unit_coverage.outputs.exists == 'true' }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: coverage-unit
+        path: tests/report/coverage-unit.xml
```

Comment:
have you try `if-no-files-found: ignore` to replace the check? https://github.com/actions/upload-artifact?tab=readme-ov-file#inputs